File: benchmarks/vm/19_value_map_loop.py
import time

t0 = time.perf_counter()
m = {}
for i in range(50_000):
    if (i % 2) == 0:
        m[f"k_{i}"] = i * 1.5
    else:
        m[f"k_{i}"] = f"v_{i}"

insert_ms = (time.perf_counter() - t0) * 1000
t1 = time.perf_counter()

num_count = 0
str_count = 0
for j in range(50_000):
    v = m[f"k_{j}"]
    if isinstance(v, float):
        num_count += 1
    elif isinstance(v, str):
        str_count += 1

read_ms = (time.perf_counter() - t1) * 1000
total_ms = insert_ms + read_ms
print(f"nums={num_count} strs={str_count}")
print(f"Insert: {int(insert_ms)} ms | Read: {int(read_ms)} ms")
print(f"BENCH_RESULT: {total_ms:.3f} ms")

File: benchmarks/vm/19_value_map_loop.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
m = {}
50_000.times do |i|
  if (i % 2) == 0
    m["k_#{i}"] = i * 1.5
  else
    m["k_#{i}"] = "v_#{i}"
  end
end

insert_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC)

num_count = 0
str_count = 0
50_000.times do |j|
  v = m["k_#{j}"]
  case v
  when Float then num_count += 1
  when String then str_count += 1
  end
end

read_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t1) * 1000
total_ms = insert_ms + read_ms
puts "nums=#{num_count} strs=#{str_count}"
puts format("Insert: %d ms | Read: %d ms", insert_ms, read_ms)
puts format("BENCH_RESULT: %.3f ms", total_ms)
